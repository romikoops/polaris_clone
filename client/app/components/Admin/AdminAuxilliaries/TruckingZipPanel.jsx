import React from 'react'
import { withNamespaces } from 'react-i18next'
import PropTypes from 'prop-types'
// import styles from '../Admin.scss'
// import { NamedSelect } from '../../NamedSelect/NamedSelect'

export const TruckingZipPanel = ({
  t, cells, rateBasis, handleRateChange, handleMinimumChange, weightSteps, currency, newCell
}) => cells.map((s, i) => {
  const wsInputs = []
  weightSteps.forEach((ws, iw) => {
    // eslint-disable-next-line react/no-array-index-key
    wsInputs.push(<div key={`ws_${iw}`} className="flex-25 layout-row layout-wrap layout-align-start-start">
      <div className="flex-100 layout-row layout-align-start-center">
        <p className="flex-none sup">{`${ws.min} - ${ws.max} ${rateBasis.label} / ${currency.label}`}</p>
      </div>
      <div className="flex-100 layout-row layout-align-start-center input_box">
        <input type="number" value={cells[i].table[iw].value} onChange={handleRateChange} name={`${i}-${iw}`} />
      </div>
    </div>)
  })

  return (
    // eslint-disable-next-line react/no-array-index-key
    <div key={`cell_${i}`} className="flex-100 layout-row layout-align-start-center layout-wrap">
      <div className="flex-50 layout-row layout-row layout-wrap layout-align-start-start">
        <p className="flex-none">{`Zipcode Range ${s.lower_zip} - ${s.upper_zip}`}</p>
      </div>
      <div className="flex-100 layout-row layout-align-start-center layout-wrap">
        <div className="flex-25 layout-row layout-wrap layout-align-start-start">
          <div className="flex-100 layout-row layout-align-start-center">
            <p className="flex-none sup">{t('admin:minimumCharge')}</p>
          </div>
          <div className="flex-100 layout-row layout-align-start-center input_box">
            <input type="number" value={s.min_value} onChange={handleMinimumChange} name={`${i}-minimum`} />
          </div>
        </div>
        {wsInputs}
      </div>
    </div>
  )
})

TruckingZipPanel.propTypes = {
  t: PropTypes.func.isRequired,
  theme: PropTypes.theme,
  cells: PropTypes.arrayOf(PropTypes.object).isRequired,
  weightSteps: PropTypes.arrayOf(PropTypes.object).isRequired,
  rateBasis: PropTypes.objectOf(PropTypes.string).isRequired,
  currency: PropTypes.objectOf(PropTypes.string).isRequired,
  handleRateChange: PropTypes.func.isRequired,
  handleMinimumChange: PropTypes.func.isRequired
}
TruckingZipPanel.defaultProps = {
  theme: {}
}
export default withNamespaces('admin')(TruckingZipPanel)
