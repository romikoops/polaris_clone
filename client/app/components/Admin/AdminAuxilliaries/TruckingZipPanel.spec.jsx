import * as React from 'react'
import { mount } from 'enzyme'
import { theme, identity } from '../../../mock'
// eslint-disable-next-line no-named-as-default
import TruckingZipPanel from './TruckingZipPanel'

const cells = [{
  lower_distance: 'FOO_LOWER_DISTANCE',
  upper_distance: 'FOO_UPPER_DISTANCE',
  table: { 0: { value: 'FOO_CELL_VALUE' } }
}]

const weightSteps = [{
  city: 'FOO_CITY',
  country: 'FOO_COUNTRY',
  max: 'FOO_MAX',
  min: 'FOO_MIN'
}]

const propsBase = {
  theme,
  cells,
  loadType: { value: 'fcl' },
  weightSteps,
  rateBasis: { label: 'FOO_RATEBASIS_LABEL' },
  currency: { label: 'FOO_CURRENCY_LABEL' },
  handleRateChange: identity,
  handleMinimumChange: identity
}

test('shallow render', () => {
  expect(mount(<TruckingZipPanel {...propsBase} />)).toMatchSnapshot()
})
