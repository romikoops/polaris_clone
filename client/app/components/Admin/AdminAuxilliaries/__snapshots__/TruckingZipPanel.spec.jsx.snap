// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render 1`] = `
<TruckingZipPanel
  cells={
    Array [
      Object {
        "lower_distance": "FOO_LOWER_DISTANCE",
        "table": Object {
          "0": Object {
            "value": "FOO_CELL_VALUE",
          },
        },
        "upper_distance": "FOO_UPPER_DISTANCE",
      },
    ]
  }
  currency={
    Object {
      "label": "FOO_CURRENCY_LABEL",
    }
  }
  handleMinimumChange={[Function]}
  handleRateChange={[Function]}
  loadType={
    Object {
      "value": "fcl",
    }
  }
  rateBasis={
    Object {
      "label": "FOO_RATEBASIS_LABEL",
    }
  }
  t={[Function]}
  theme={
    Object {
      "colors": Object {
        "primary": "#333",
        "secondary": "#fafafa",
      },
    }
  }
  weightSteps={
    Array [
      Object {
        "city": "FOO_CITY",
        "country": "FOO_COUNTRY",
        "max": "FOO_MAX",
        "min": "FOO_MIN",
      },
    ]
  }
>
  <div
    className="flex-100 layout-row layout-align-start-center layout-wrap"
    key="cell_0"
  >
    <div
      className="flex-50 layout-row layout-row layout-wrap layout-align-start-start"
    >
      <p
        className="flex-none"
      >
        Zipcode Range undefined - undefined
      </p>
    </div>
    <div
      className="flex-100 layout-row layout-align-start-center layout-wrap"
    >
      <div
        className="flex-25 layout-row layout-wrap layout-align-start-start"
      >
        <div
          className="flex-100 layout-row layout-align-start-center"
        >
          <p
            className="flex-none sup"
          >
            Minimum charge (Flat Rate)
          </p>
        </div>
        <div
          className="flex-100 layout-row layout-align-start-center input_box"
        >
          <input
            name="0-minimum"
            onChange={[Function]}
            type="number"
          />
        </div>
      </div>
      <div
        className="flex-25 layout-row layout-wrap layout-align-start-start"
        key="ws_0"
      >
        <div
          className="flex-100 layout-row layout-align-start-center"
        >
          <p
            className="flex-none sup"
          >
            FOO_MIN - FOO_MAX FOO_RATEBASIS_LABEL / FOO_CURRENCY_LABEL
          </p>
        </div>
        <div
          className="flex-100 layout-row layout-align-start-center input_box"
        >
          <input
            name="0-0"
            onChange={[Function]}
            type="number"
            value="FOO_CELL_VALUE"
          />
        </div>
      </div>
    </div>
  </div>
</TruckingZipPanel>
`;
